### DISCUSSION TITLE
Test latest backups with CopyFile param using striped backups

### DISCUSSION THREAD

**Discussion by mchuc7719** [dataplat/dbatools#8865] [Q&A] [answered]
I am using latest dbatools version 1.1.146.
I am trying to test latest backups using CopyFile param, the Full backup it is striped in 4 files using SQL Server 2012 Device Backup(files).

Afte some debug process I realized this code(allcomands.ps1):
$lastbackup = Get-DbaDbBackupHistory -SqlInstance $sourceserver -Database $dbName -IncludeCopyOnly:$IncludeCopyOnly -Last -DeviceType $DeviceType -WarningAction SilentlyContinue -IgnoreDiffBackup:$IgnoreDiffBackup

The object $lastbackup has the names of the striped backup files, I mean it is an array:

![testlastdba](https://user-images.githubusercontent.com/43049627/232254247-0a194b87-b455-46bc-8c15-10ff65b9a0de.png)

Then when the process tries to get the filename in order to copy, it gets the name in a wrong name, gets the all file names in one liine:

![testlastdba-2](https://user-images.githubusercontent.com/43049627/232254263-34d99f2e-c360-4d48-8f64-cb1b67d6bed5.png)

This only happens when I use Copyfile, without this param, the process runs like charms!

Please any advice about how to use the param copy file when the backup files are striped.

Thanks in advance.

**Comment by potatoqualitee** [comment]
Thanks for this detail, ill take a look

**Reply by mchuc7719** [reply]
Hi Potatoqualitee,

Please, any advise or workaround?.

On my side, I wrote a small script to copy all files before to run Test-DbaLastBackup, because all the process(without copy first) takes around 14 hrs.

Regards.

**Reply by potatoqualitee** [reply, marked as answer]
hey @mchuc7719  -- the fix is in dbatools 2.0.0-preview7

https://github.com/dataplat/dbatools/pull/8868

Please note that 2.0 has a bunch of changes which you can read about more at my blog https://blog.netnerds.net (there's like 3 diff posts that talk about it)
